src/test/kotlin/com/zepben/evolve/cim/iec61970/base/wires/RegulatingControlTest.kt

```diff
@@ -0,0 +1,75 @@
+
+import com.zepben.evolve.cim.iec61970.base.core.PhaseCode
+import com.zepben.evolve.services.network.NetworkService
+import com.zepben.evolve.services.network.testdata.fillFields
+import com.zepben.evolve.utils.PrivateCollectionValidator
+import com.zepben.testutils.junit.SystemLogExtension
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+
+class RegulatingControlTest {
+
+    @JvmField
+    @RegisterExtension
+    var systemErr: SystemLogExtension = SystemLogExtension.SYSTEM_ERR.captureLog().muteOnSuccess()
+
+    @Test
+    internal fun constructorCoverage() {
+        assertThat(object : RegulatingControl() {}.mRID, Matchers.not(equalTo("")))
+        assertThat(object : RegulatingControl("id") {}.mRID, equalTo("id"))
+    }
+
+    @Test
+    internal fun accessorCoverage() {
+        val regulatingControl = object : RegulatingControl() {}
+
+        assertThat(regulatingControl.discrete, nullValue())
+        assertThat(regulatingControl.enabled, nullValue())
+        assertThat(regulatingControl.mode, equalTo(RegulatingControlModeKind.UNKNOWN_CONTROL_MODE))
+        assertThat(regulatingControl.monitoredPhase, equalTo(PhaseCode.NONE))
+        assertThat(regulatingControl.targetDeadband, nullValue())
+        assertThat(regulatingControl.targetValue, nullValue())
+        assertThat(regulatingControl.enabled, nullValue())
+        assertThat(regulatingControl.maxAllowedTargetValue, nullValue())
+        assertThat(regulatingControl.minAllowedTargetValue, nullValue())
+        assertThat(regulatingControl.terminal, nullValue())
+
+        regulatingControl.fillFields(NetworkService(), true)
+
+        assertThat(regulatingControl.discrete, equalTo(false))
+        assertThat(regulatingControl.mode, equalTo(RegulatingControlModeKind.voltage))
+        assertThat(regulatingControl.monitoredPhase, equalTo(PhaseCode.ABC))
+        assertThat(regulatingControl.targetDeadband, equalTo(2.0f))
+        assertThat(regulatingControl.targetValue, equalTo(100.0))
+        assertThat(regulatingControl.enabled, equalTo(true))
+        assertThat(regulatingControl.maxAllowedTargetValue, equalTo(200.0))
+        assertThat(regulatingControl.minAllowedTargetValue, equalTo(50.0))
+        assertThat(regulatingControl.terminal, notNullValue())
+    }
+
+    @Test
+    internal fun regulatingCondEqs() {
+        PrivateCollectionValidator.validate(
+            { object : RegulatingControl() {} },
+            { id, _ -> object : RegulatingCondEq(id) {} },
+            RegulatingControl::numRegulatingCondEqs,
+            { rc, mRID -> rc.getRegulatingCondEq(mRID) },
+            RegulatingControl::regulatingCondEqs,
+            RegulatingControl::addRegulatingCondEq,
+            RegulatingControl::removeRegulatingCondEq,
+            RegulatingControl::clearRegulatingCondEqs
+        )
+    }
+}
```

Comment:
Add trailing newline (applies to every new file)

Reply:
protip if you've got ripgrep:
```rg -Ul '[^\n]\z'```

Reply:
fixed across the whole project